engine/execution/state_synchronization/state_diff_storer_test.go

```diff
@@ -0,0 +1,134 @@
+package state_synchronization
+
+import (
+	"context"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/fxamacker/cbor/v2"
+	datastore "github.com/ipfs/go-datastore/examples"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+
+	"github.com/onflow/flow-go/engine/execution/computation/computer/uploader"
+	cborcodec "github.com/onflow/flow-go/model/encoding/cbor"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/network/compressor"
+)
+
+func makeBlockstore(t *testing.T, name string) (blockstore.Blockstore, func()) {
+	dsDir := filepath.Join(os.TempDir(), name)
+	require.NoError(t, os.RemoveAll(dsDir))
+	err := os.Mkdir(dsDir, fs.ModeDir)
+	require.NoError(t, err)
+
+	ds, err := datastore.NewDatastore(dsDir)
+	require.NoError(t, err)
+
+	return blockstore.NewBlockstore(ds.(*datastore.Datastore)), func() {
+		require.NoError(t, os.RemoveAll(dsDir))
+	}
+}
+
+const BUCKET_NAME = "flow_public_mainnet14_execution_state"
```

Comment:
This is ok for local testing (as currently intended), but we'll definitely want to avoid pulling lots of data from GCP for a ci test since that will add up on our GCP bill